lib/decorators/component.ts

```diff
@@ -192,50 +200,42 @@ export function View(
 	}
 }
 
-// ## Component Provider Parser
 Module.addProvider(TYPE, (target: any, name: string, injects: string[], ngModule: ng.IModule) => {
 	// First create an empty object to contain the directive definition object
 	let ddo: any = {};
 
-	componentStore.forEach((val, key) => {
-		// Loop through the key/val pairs of metadata and assign it to the DDO
-		ddo[key] = val;
-	}, target);
+	// Loop through the key/val pairs of metadata and assign it to the DDO
+	componentStore.forEach((val, key) => ddo[key] = val, target);
```

Comment:
I know this is a piece of old code, but do you think this functionality should be moved into the Metastore class?